addons/l10n_in/models/company.py

```diff
@@ -12,18 +14,43 @@ class ResCompany(models.Model):
             ("8", "8 Digits"),
         ],
         string="HSN Code Digit",
-        compute="_compute_l10n_in_hsn_code_digit",
+        compute="_compute_l10n_in_hsn_code_digit_and_l10n_in_pan",
         store=True,
         readonly=False,
     )
+    l10n_in_pan = fields.Char(
+        string="PAN",
+        compute="_compute_l10n_in_hsn_code_digit_and_l10n_in_pan",
+        store=True,
+        readonly=False,
+        help="PAN enables the department to link all transactions of the person with the department.\n"
+             "These transactions include taxpayments, TDS/TCS credits, returns of income/wealth/gift/FBT,"
+             "specified transactions, correspondence, and so on.\n"
+             "Thus, PAN acts as an identifier for the person with the tax department.",
+    )
+    l10n_in_pan_type = fields.Char(string="PAN Type", compute="_compute_l10n_in_pan_type")
 
     @api.depends('vat')
-    def _compute_l10n_in_hsn_code_digit(self):
+    def _compute_l10n_in_hsn_code_digit_and_l10n_in_pan(self):
         for record in self:
-            if record.vat:
+            if record.country_code == "IN" and record.vat:
                 record.l10n_in_hsn_code_digit = "4"
+                record.l10n_in_pan = record.vat[2:12] if gstin.is_valid(record.vat) else False
```

Comment:
might be worth using this: https://github.com/arthurdejong/python-stdnum/blob/master/stdnum/in_/gstin.py#L145C5-L145C11